SConstruct

```diff
@@ -136,14 +136,19 @@ def RecursiveGlob(pattern, dir_name=buildDirectory):
 	# Start with source files in subdirectories.
 	matches = [RecursiveGlob(pattern, sub_dir) for sub_dir in Glob(pathjoin(str(dir_name), "*"))
 		if isinstance(sub_dir, Dir)]
-	# Add source files in this directory, except for main.cpp
+	# Add source files in this directory, except for main.cpp and support library code.
 	matches += Glob(pathjoin(str(dir_name), pattern))
-	matches = [i for i in matches if not '{}main.cpp'.format(os.path.sep) in str(i)]
+
+	transform = lambda s : ('{}' + s).format(os.path.sep)
+	matches = [i for i in matches if not transform("main.cpp") in str(i)]
+	matches = [i for i in matches if not transform("render") in str(i)]
+	matches = [i for i in matches if not "Shader.cpp" in str(i)]
+	matches = [i for i in matches if not transform("GameWindow.cpp") in str(i)]
```

Comment:
minor comment: the code here now loops through matches four times. I think it should be possible to run through the matches just once (with the 4 criteria in the if-part).